File: php/features.php
<?php
declare(strict_types=1);

// RealTimeBusData SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RealTimeBusDataFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RealTimeBusDataBaseFeature();
            case "test":
                return new RealTimeBusDataTestFeature();
            default:
                return new RealTimeBusDataBaseFeature();
        }
    }
}

File: py/features.py
# RealTimeBusData SDK feature factory

from feature.base_feature import RealTimeBusDataBaseFeature
from feature.test_feature import RealTimeBusDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RealTimeBusDataBaseFeature(),
        "test": lambda: RealTimeBusDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
